File: ph.kana.memory.stash/test/ph/kana/memory/codec/PasswordCodecTest.java
package ph.kana.memory.codec;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class PasswordCodecTest {

	@Test
	void encrypt_decrypt_symmetrical() throws Exception {
		String rawPassword = "password123";
		String salt = Long.toString(System.currentTimeMillis());
		PasswordCodec codec = new PasswordCodec();

		EncryptedPassword encrypted = codec.encrypt(rawPassword, salt);
		String decrypted = new String(codec.decrypt(encrypted, salt));

		assertEquals(rawPassword, decrypted);
	}
}

File: memory.stash/src/test/java/ph/kana/memory/codec/PasswordCodecTest.kt
package ph.kana.memory.codec

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test


class PasswordCodecTest {

    @Test
    fun `decrypt should be able to return the original encrypted text`() {
        val codec = PasswordCodec()

        val original = "password123"
        val salt = "salt"
        val encrypted = codec.encrypt(original, salt)

        val decrypted = codec.decrypt(encrypted, salt)

        assertEquals(String(decrypted), original)
    }
}